src/main/kotlin/org/stellar/walletsdk/util/Sep24InteractiveFlow.kt

```diff
@@ -0,0 +1,139 @@
+ * @param httpClient HTTP client
+ *
+ * @return response object from the anchor
+ *
+ * @throws [AssetNotAcceptedForDepositException] if asset is not accepted for deposits
+ * @throws [AssetNotEnabledForDepositException] if asset is not enabled for deposits by the anchor
+ * @throws [AssetNotAcceptedForWithdrawalException] if asset is not accepted for withdrawals
+ * @throws [AssetNotEnabledForWithdrawalException] if asset is not enabled for withdrawals by the
+ * anchor
+ * @throws [NetworkRequestFailedException] if network request fails
+ */
+suspend fun interactiveFlow(
+  type: InteractiveFlowType,
+  accountAddress: String,
+  fundsAccountAddress: String? = null,
+  homeDomain: String,
+  assetCode: String,
+  memoId: String? = null,
+  clientDomain: String? = null,
+  extraFields: Map<String, Any>? = null,
+  walletSigner: WalletSigner,
+  anchor: Anchor,
+  server: Server,
+  network: Network,
+  httpClient: OkHttpClient,
+): InteractiveFlowResponse {
+  return CoroutineScope(Dispatchers.IO)
+    .async {
+      val sep24RequiredFields =
+        listOf(
+          StellarTomlFields.SIGNING_KEY.text,
+          StellarTomlFields.TRANSFER_SERVER_SEP0024.text,
+          StellarTomlFields.WEB_AUTH_ENDPOINT.text
+        )
+      val toml = StellarToml(homeDomain, server, httpClient)
+      val tomlContent = toml.getToml()
+
+      // Check toml for required SEP-24 fields
+      val missingFields = mutableListOf<String>()
+
+      sep24RequiredFields.forEach { field ->
+        if (tomlContent[field] == null) {
+          missingFields.add(field)
+        }
+      }
+
+      if (missingFields.size > 0) {
+        throw StellarTomlMissingFields(missingFields)
+      }
+
+      val transferServerEndpoint =
+        tomlContent[StellarTomlFields.TRANSFER_SERVER_SEP0024.text].toString()
+
+      val serviceInfo = anchor.getServicesInfo(transferServerEndpoint)
+
+      // Check if deposit/withdraw is enabled for the asset
+      if (type == InteractiveFlowType.DEPOSIT) {
+        val asset =
+          serviceInfo.deposit[assetCode] ?: throw AssetNotAcceptedForDepositException(assetCode)
+
+        if (!asset.enabled) {
+          throw AssetNotEnabledForDepositException(assetCode)
+        }
+      } else {
+        val asset =
+          serviceInfo.withdraw[assetCode] ?: throw AssetNotAcceptedForWithdrawalException(assetCode)
+
+        if (!asset.enabled) {
+          throw AssetNotEnabledForWithdrawalException(assetCode)
+        }
+      }
+
+      // SEP-10 auth
+      val auth =
+        Auth(
+          accountAddress = accountAddress,
+          webAuthEndpoint = tomlContent[StellarTomlFields.WEB_AUTH_ENDPOINT.text].toString(),
+          homeDomain = homeDomain,
+          clientDomain = clientDomain,
+          memoId = memoId,
+          networkPassPhrase = network.networkPassphrase,
+          walletSigner = walletSigner,
+          httpClient = httpClient
+        )
+      val authToken = auth.authenticate()
```

Comment:
What if the wallet has already authenticated and has their own `authToken`? We should allow them to pass it instead of forcing them to go through SEP-10 again.

Reply:
Or, we take the same approach as the toml contents and cache the auth token on the `Anchor` class.

Reply:
Yup! This is related to caching toml file. We will try to find a solution in the status update PR.